library.py

```diff
@@ -177,15 +177,16 @@ def adjust_linker_offset(script_name, ldscript):
             r"\1+%s\2-%s" % (offset_address, offset_address),
             content, flags=re.MULTILINE)
 
-    offset_script = os.path.join(env.subst('$PROJECTBUILD_DIR'),
+    offset_script = os.path.join(env.subst('$PROJECT_BUILD_DIR'),
                     "%s_flash_%s.ld" % (script_name, offset_address))
 
     with open(offset_script, "w") as fp:
         fp.write(content)
 
     return offset_script
 
-linker_script = adjust_linker_offset(env.subst('$PIOENV'), linker_script)
+# following function does not work with custom board due to platformio bug
+# linker_script = adjust_linker_offset(env.subst('$PIOENV'), linker_script)
```

Comment:
The solution to failure with a custom board cannot be to disable this important feature. This works for non-custom boards and it specifically prevents the problem you mention in the readme of overwriting the bootloader.

Please detect the use of a custom board and disable the adjustment only in that case.

Reply:
agree, enabled it again.
Currently didn't catch the error, which will make it a good example to fix the underlying issue: https://github.com/platformio/platformio-core/issues/3264

Reply:
this issue is now resolved!